plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/EntandoCrudService.java

```diff
@@ -0,0 +1,18 @@
+package org.entando.entando.plugins.jacms.aps.system.services;
+
+import org.entando.entando.web.common.model.*;
+
+import java.util.Optional;
+
+public interface EntandoCrudService<T, ID> {
```

Comment:
I like the effort to make services more homogeneous, however it is not easy to apply this interface on the current structure of the code. For example in the `ContentModelService` the `Optional<T> findById(ID id);` is used only internally in the service implementation because it is the service itself that throws the Resource Not Found exception.
Moreover in `ContentTypeService` `update()` and `create()` methods need also the `BindingResult` parameters.
I think that we should decide if we want to keep this interface and made some additional efforts to use it (then it should also be moved in core, since it has a quite general purpose) or if we are not going to use it.

Reply:
I'll remove this. I started it as a way of unifying services with DB persistence to have an interface that would make it easier to migrate to a Spring Repository later if we decided to. That was before I found out about the Entity classes that use a completely different structure and persistence and when I had multiple classes with a common DB persistence structure. Now that the code was refactored to use the existing Entity structure I reckon that is better to delete this and  leave this kind of change to a story that deals specifically with changes related to DB persistence

Reply:
Interface removed